x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -243,6 +243,25 @@ export class JobCreator {
     );
   }
 
+  public set annotations(enable: boolean) {
+    if (enable) {
+      if (this._job_config.model_plot_config) {
+        this._job_config.model_plot_config.annotations_enabled = true;
+      } else {
+        this._job_config.model_plot_config = {
+          enabled: false,
```

Comment:
i believe this isn't needed, `enabled` should default to `false`

Reply:
both this and `public set modelPlot` above should have a guard that `model_plot_config` is deleted if all items inside are false.
line 235 above (thanks github for not allowing me to add a comment there) should not now delete `model_plot_config` if `modelPlot` is set to false as it will wipe out `annotations_enabled`

Reply:
I tested trying to create a job where model plot was disabled, but model change annotations were enabled, and if I check in the JSON for the created job, this block which should be there is missing:
```
  "model_plot_config": {
    "annotations_enabled": true
  },
```

Reply:
I think I had to do `enabled: false` there because ModelPlotConfig interface has enabled as required. And you're right James, deleting that was causing the bug Pete ran into. I have updated to delete the `model_plot_config` object only if both of the `enabled` and `annotations_enabled` are false in https://github.com/elastic/kibana/pull/70678/commits/1b54be9b39e990830c85f2506d040e803725dad5.